control-plane/gateways/config.go

```diff
@@ -38,3 +41,8 @@ type GatewayConfig struct {
 	// defined on a Gateway.
 	MapPrivilegedServicePorts int
 }
+
+type GatewayResources struct {
```

Comment:
Both gateway resources and gateway cleanup need this struct so I pulled it into the gateways package for ease of use

Reply:
good call, I like this change